Урок 2
Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона
Домашнее задание

Задача 1

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).

In [2]:
import numpy as np
def combinations(n: int, k: int) -> int:
    """Число сочетаний"""
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k))

In [3]:
# Пусть X - дискретная случайная величина, равная числу правильных ответов. 
# Она распределена по биномиальному закону с параметрами n = 5 (число вопросов),
# p = 1/4 (вероятность угадать ответ), поэтому вероятности вычисляются по формуле Бернулли: 
x_values = np.arange(6)
x0 = combinations(5, 0)*((3/4)**5) #= 243/1024
x1 = combinations(5, 1)*((3/4)**4)*(1/4) #= 405/1024 
x2 = combinations(5, 2)*((3/4)**3)*((1/4)**2) #= 270/1024
x3 = combinations(5, 3)*((3/4)**2)*((1/4)**3) #= 90/1024
x4 = combinations(5, 4)*(3/4)*((1/4)**4) #= 15/1024
x5 = combinations(5, 5)*((1/4)**5) #= 1/1024
# Проверим:
print((243+405+270+90+15+1)/1024)

1.0


In [4]:
# Найдем M(X):
m = 0 + 405/1024 + 2*(270/1024) + 3*(90/1024) + 4*(15/1024) + 5/1024
print(m)

1.25


In [5]:
# Найдем D(X):
y_values = x_values - m
print(y_values)
#Y = [-1.25, -0.25, 0.75, 1.75, 2.75, 3.75]
z_values = y_values ** 2
print(z_values)
#Z = [1.5625, 0.0625, 0.5625, 3.0625, 7.5625, 14.0625] 
d = z_values[0]*x0 + z_values[1]*x1 + z_values[2]*x2 + z_values[3]*x3+ z_values[4]*x4 + z_values[5]*x5
print(d)

[-1.25 -0.25  0.75  1.75  2.75  3.75]
[ 1.5625  0.0625  0.5625  3.0625  7.5625 14.0625]
0.9375


Ответ: закон распределения числа правильных ответов при простом угадывании: 
{0: 243/1024, 1: 405/1024, 2: 270/1024, 3: 90/1024, 4: 15/1024, 5: 1/1024},
    M(X) = 1,25,
    D(X) = 0,9375

Задача 2

Пользователь получает в среднем 10 писем со спамом на свой почтовый ящик за сутки. Найти число N, такое, что с вероятностью 0.95 пользователь получит не более N писем со спамом за текущий день.

In [6]:
def poisson_proba(k: int, lambda_: float) -> float:
    """Формула Пуассона"""
    return (lambda_ ** k) * (np.exp(-lambda_)) / np.math.factorial(k)

In [7]:
# Мы знаем результат - суммарную вероятность того, что пользователь получит от 0 до N писем, и среднее значение:
lambda_ = 10

In [8]:
# Так как количество писем может быть только натуральным числом, можем найти минимальное значение, при котором 
# суммарная вероятность получить писем не более этого количества будет 0,95: 
for n in range(100):
    p = sum(poisson_proba(k=i, lambda_=lambda_) for i in range(n+1))
    #print(f'Вероятность получить не более {n} писем равна {p}')
    if p >= 0.95:
        print(n)
        break

15


Ответ: с вероятностью 0.95 пользователь получит не более 15 писем со спамом за текущий день.

Задача 3

Производятся выстрелы по мишени. Вероятность попадания в мишень при одном выстреле равна 0.01. Сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 6 раз будет совершено попадание?

Подсказка. Здесь предстоит немножко покодить.

In [16]:
def combinations(n: int, k: int) -> int:
    """Число сочетаний"""
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k))

In [24]:
# применим формулу Бернулли:
# p = 0.01
# k = 6
# P(X>=6) = 0.9  (хотя бы 6 раз - значит, 6 и больше, cоответственно, P(X<6) = 0.1)
# найдем значение n:
for n in range(6, 2000):
    pq = sum(combinations(n, k)*(0.01**k)*(0.99**(n-k)) for k in range(0, 6))
    if pq <= 0.1:
        print(n)
        break

926


Ответ:  чтобы быть уверенным с вероятностью 0.9, что хотя бы 6 раз будет совершено попадание, нужно выстрелить 926 раз.